eth_abi/grammar.py

```diff
@@ -185,6 +185,18 @@ def invalidate(self, error_msg):
             ),
         )
 
+    @property
+    def is_array(self):
+        return self.arrlist is not None
+
+    @property
+    def is_dynamic(self):
+        raise NotImplementedError('Must implement `is_dynamic`')
+
+    @property
+    def has_dynamic_arrlist(self):
```

Comment:
Does `has_dynamic_arrlist` need to be public? Seems like it's mostly a helper property, so my instinct is to `_`-prefix it until we know we want to use it somewhere.